### ISSUE TITLE
.12 templates single quotes

### ISSUE THREAD

**Issue by wburn-sierra-cedar** [hashicorp/terraform#24482]
### Terraform Version

```
Terraform v0.12.24
+ provider.aws v2.51.0
+ provider.null v2.1.2
+ provider.random v2.2.1
+ provider.template v2.1.2
```

### Terraform Configuration Files

```hcl
locals {
config_web_json = templatefile(var.customization, 
   {<vars>})
} 
template:
<SNIP>
ps_cust_home:
   location:  "/custom/%{hiera('db_service_name')}"
</SNIP>
```

### Crash Output

customizations.yaml:35,32-33:
Invalid character; Single quotes are not valid. Use double quotes (") to
enclose strings., and 2 other diagnostic(s).

### Expected Behavior

This template worked fine in .11

we can't change the syntax of the template to not use single quotes or percent sign.

Also if I do change the quotes to double, I get errors about % symbol.  I tried escaping out the quotes and %.  How do I use these characters in a template as was possible in .11?

**Comment by danieldreier** [comment]
@wburn-sierra-cedar I'm glad you're making the jump to 0.12! It can be a lot of work, but it's worth doing. 0.12 includes breaking changes, and not everything that worked in 0.11 is expected to work in 0.12. 

Fortunately, this specific instance looks relatively straightforward for you to fix. The [expressions documentation](https://www.terraform.io/docs/configuration/expressions.html#string-literals) documents that `$$` and `%%` work to get literals of those characters.

I just tested this to make sure, and the following works:

example.tmpl
```
$$ foo %%
```
repro.tf
```terraform
resource "local_file" "example_file" {
    content     = templatefile("${path.module}/example.tmpl", {})
    filename = "/tmp/example.txt"
}
```
contents of `/tmp/example.txt` after a terraform 0.12.24 apply:
```
$$ foo %%
```

It looks to me like you're generating Puppet hiera configs in YAML. If you're struggling with consistently producing valid YAML in terraform templates, you might also play with the [yamlencode](https://www.terraform.io/docs/configuration/functions/yamlencode.html) function. I can't tell you whether it's worth the trade-offs, but I wanted to point out that it exists.

Based on the information you've provided, I don't immediately see evidence of a bug in Terraform core, so I'm going to close this issue. If the suggestions I made don't solve your issue, your best bet is to seek support on the community forum. We use GitHub issues for tracking bugs and enhancements, rather than for questions. While we can sometimes help with certain simple problems here, it's better to use [the community forum](https://discuss.hashicorp.com/c/terraform-core) where there are more people ready to help. The GitHub issues here are monitored only by our few core maintainers.
